test/Dialect/FIRRTL/extract-instances.mlir

```diff
@@ -65,17 +65,27 @@ firrtl.circuit "ExtractBlackBoxesSimple" attributes {annotations = [{class = "fi
 
 // CHECK: firrtl.circuit "ExtractBlackBoxesSimple2"
 firrtl.circuit "ExtractBlackBoxesSimple2" attributes {annotations = [{class = "firrtl.transforms.BlackBoxTargetDirAnno", targetDir = "BlackBoxes"}]} {
-  // CHECK: firrtl.hierpath @nla_1{{.*}} [@ExtractBlackBoxesSimple2::@bb, @MyBlackBox]
-  // CHECK-NOT: firrtl.hierpath @nla_2
-  // CHECK-NOT: firrtl.hierpath @nla_3
-  firrtl.hierpath @nla_1 [@BBWrapper::@bb, @MyBlackBox]
-  firrtl.hierpath @nla_2 [@DUTModule::@mod, @BBWrapper::@bb]
-  firrtl.hierpath @nla_3 [@ExtractBlackBoxesSimple2::@dut, @DUTModule::@mod, @BBWrapper::@bb]
+  // Old style NLAs
+  firrtl.hierpath @nla_old1 [@DUTModule::@mod, @BBWrapper::@bb]
+  firrtl.hierpath @nla_old2 [@ExtractBlackBoxesSimple2::@dut, @DUTModule::@mod, @BBWrapper::@bb]
+  // New style NLAs on extracted instance
+  firrtl.hierpath @nla_on1 [@DUTModule::@mod, @BBWrapper]
+  firrtl.hierpath @nla_on2 [@ExtractBlackBoxesSimple2::@dut, @DUTModule::@mod, @BBWrapper]
```

Comment:
We might want to enforce the new style the same way we do for wires/regs. That would mean a hierpath always ends at the parent module, irrespective of the op, and the last element of a hierpath is alwyas a `SymbolRef`.

Otherwise we again have this degree of freedom where an NLA on an instance *could* mention the instance name in the hierpath, but it wouldn't have to. Forcing one or the other should allow for simpler implementations :thinking:.

Another benefit would be that only hierpaths passing *through* an instance for the sake of addressing something in the subhierarchy actually mentions the instance in the hierpath. Annos that are *on* the instance would have the path terminate earlier, making updating easier.